### ISSUE TITLE
Water Powered Saw CraftTweaker Issues

### ISSUE THREAD

**Issue by Vyraal1** [vadis365/PrimalTech#27]
Script here: https://hastebin.com/boxeragagu.coffeescript

The recipe is recognized in JEI, but putting a plank down on the water saw actually does nothing.

![image](https://user-images.githubusercontent.com/26587049/43750802-16135b8e-99c8-11e8-94cb-488b835308b6.png)

![image](https://user-images.githubusercontent.com/26587049/43750816-23e6acf2-99c8-11e8-8757-05e73bc8f4a0.png)

Also was just wondering, it says the master branch is ahead of the 1.12 branch, but I wasn't able to find the CraftTweaker compat there at all (only on the 1.12 branch), wasn't sure if this was intentional or not.

**Comment by vadis365** [comment]
Hello,
Ignore the git branches - the 1.12 branch is the only one that is important, the mod was started back in 1.10.2 when SevTech dev began - any other versions 1.10.2/1.11.2 or whatever were in dev versions for the pack dev.
When you say 'does nothing' do you mean no animation, no block functions at all, or do you mean it cuts the block and produces no output? If the recipe is in JEI then it has been registered; as the crafttweaker wrapper uses the internal recipe class from the mod. The water saw will always turn and show an animation if it is placed over water correctly - if there is no animation it's not.
Cheers,
Me.

**Comment by Vyraal1** [comment]
So placing the planks down on the Water Saw still shows the saw blade spinning, however the plank does not move at all across the saw or get cut (waited a full minute or so and there was still no movement on the plank).

The water saw will cut it's default recipes just fine, such as cutting logs into planks.

**Comment by vadis365** [comment]
did you do a restart after adding the recipe?

**Comment by vadis365** [comment]
Hello,
Would it be possible for you to add the exact same recipe to the mod's own config for recipes and tell me if it functions outside the craft tweaker script for what you need? I know this is a pain as it involves a re-start, but it would help to know if the base mod or the CT integration is the issue here.
Cheers,
Me.

**Comment by Vyraal1** [comment]
Yes, checked for `/ct syntax`, restarted. I can confirm I had to use the configs to successfully add the recipe, but would prefer zenscript as I wanted to use arrays.
![image](https://user-images.githubusercontent.com/26587049/43961156-2bd71e80-9c82-11e8-9d66-4f6c722f1f6e.png)

Configs: https://hastebin.com/zadovemagi.coffeescript

**Comment by vadis365** [comment]
Hello,
I'll take a look at the code and see where stuff is wonky, thanks for the bug report.
On a side note - you disable my absorption hopper in MGU - feelsBadMan - hehe.
Cheers,
Me.

**Comment by Vyraal1** [comment]
Hey Vadis, found some other stuff that might help you track down the issue here, so far I think it's probably related to how ore dictionaries are handled. 

I used the following script to add a cheese recipe to the Wooden Basin. `/ct syntax` showed no errors, and no errors in the crafttweaker log. However when I checked the recipes of the Wooden Basin in game, there was nothing, not even the default obsidian recipe. I decided to to put milk + salt in the Wooden Basin and try it out anyways and got the following crash report.

**Script:**: https://pastebin.com/YAbRGtZj
**Crash Report**: https://pastebin.com/XBTUeKKF
